src/expressions/visitors/TimeIndexVisitor.cpp

```diff
@@ -132,9 +144,37 @@ TimeIndex TimeIndexVisitor::visit([[maybe_unused]] const Nodes::AllTimeSumNode*
     return TimeIndex::CONSTANT_IN_TIME_AND_SCENARIO;
 }
 
-TimeIndexVisitor::TimeIndexVisitor(std::unordered_map<const Nodes::Node*, TimeIndex> context):
-    context_(std::move(context))
+TimeIndexVisitor::TimeIndexVisitor(std::unordered_map<const Nodes::Node*, TimeIndex> context,
+                                   const std::string& componentId,
+                                   const std::vector<Connection>& connections):
+    context_(std::move(context)),
+    componentId_(componentId),
+    connections_(connections)
+{
+}
+
+TimeIndexVisitor::TimeIndexVisitor(const std::string& componentId,
+                                   const std::vector<Connection>& connections):
+    componentId_(componentId),
+    connections_(connections)
+{
+}
+
+std::vector<const Component*> TimeIndexVisitor::getConnectedComponents()
```

Comment:
duplicated code!

Reply:
I removed this duplication : I opted to supply a component with a  list of other components connected to it.
Indeed, when all system components and connections are read, we actually connect all components by the above list.